app/src/test/java/org/mozilla/fenix/search/toolbar/ToolbarViewTest.kt

```diff
@@ -158,6 +160,161 @@ class ToolbarViewTest {
         verify { editToolbar.setIcon(any(), "Search Engine") }
     }
 
+    @Test
+    fun `GIVEN a general search engine is default WHEN a topic specific engine is selected THEN the hint changes`() {
+        val generalEngine = SearchEngine(
+            id = UUID.randomUUID().toString(),
+            name = "General",
+            icon = testContext.getDrawable(R.drawable.ic_search)!!.toBitmap(),
+            type = SearchEngine.Type.BUNDLED,
+            resultUrls = listOf(
+                "https://www.startpage.com/sp/search?q={searchTerms}",
+            ),
+            isGeneral = true,
+        )
+        val topicSpecificEngine = SearchEngine(
+            id = UUID.randomUUID().toString(),
+            name = "TopicSpecific",
+            icon = testContext.getDrawable(R.drawable.ic_search)!!.toBitmap(),
+            type = SearchEngine.Type.BUNDLED,
+            resultUrls = listOf(
+                "https://www.startpage.com/sp/search?q={searchTerms}",
+            ),
+            isGeneral = false,
+        )
+        val toolbarView = buildToolbarView(false)
+        toolbarView.update(defaultState.copy(searchEngineSource = SearchEngineSource.Default(generalEngine)))
```

Comment:
nit: Looks like there is already a `general` search engine defined for the `defaultState` so we could just use that with `view.update(defaultState)`?